cardano-submit-api/cardano-submit-api.cabal

```diff
@@ -0,0 +1,145 @@
+cabal-version: 2.2
+
+name:                   cardano-submit-api
+version:                3.1.2
+synopsis:               A web server that allows transactions to be POSTed to the cardano chain
+description:
+homepage:               https://github.com/input-output-hk/cardano-node
+bug-reports:            https://github.com/input-output-hk/cardano-node/issues
+license:                Apache-2.0
+license-file:           LICENSE
+author:                 IOHK Engineering Team
+maintainer:             operations@iohk.io
+copyright:              (c) 2019-2021 IOHK
+category:               Language
+build-type:             Simple
+extra-source-files:     CHANGELOG.md
+
+common base                         { build-depends: base                             >= 4.12       && < 4.15     }
```

Comment:
Is this some new cabal feature I am not aware of?

Reply:
I think this has existed since `cabal-3.2`.  It lets you re-use config.

Here, I'm ensuring that all the components in the cabal file share the same bounds.  I've also grouped some default config into on `common project-config` stanza so as to ensure all the components share the same warning settings.